dashboard/app/models/plc/enrollment_unit_assignment.rb

```diff
@@ -75,13 +75,8 @@ def check_for_unit_completion
   def enroll_in_module(learning_module)
     return unless learning_module.plc_course_unit == plc_course_unit
 
-    module_assignment = Plc::EnrollmentModuleAssignment.find_or_create_by(plc_enrollment_unit_assignment: self, plc_learning_module: learning_module)
-    learning_module.plc_tasks.each do |task|
-      Plc::EnrollmentTaskAssignment.find_or_create_by(
-          plc_enrollment_module_assignment: module_assignment,
-          plc_task: task,
-          status: Plc::EnrollmentTaskAssignment::NOT_STARTED
-      )
-    end
+    Plc::EnrollmentModuleAssignment.find_or_create_by(plc_enrollment_unit_assignment: self,
+                                                      plc_learning_module: learning_module,
+                                                      user: user)
```

Comment:
I don't have a good route to test this without an evaluation in my development env.